bin/run-samples

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env ruby
+
+require 'open3'
+require 'thread'
+
+$signals = Queue.new
```

Comment:
I was doing a lot of hacking around on the loop early on, and the global was expedient. Will definitely revise to something more tidy.